src/lib/styles/stylesheet-processor.ts

```diff
@@ -42,10 +41,25 @@ export class StylesheetProcessor {
       browserslistData: this.browserslistData,
     };
 
-    const { css, warnings } = this.styleSheetProcessorWorker(workerOptions);
+    const ioChannel = new MessageChannel();
 
-    warnings.forEach(msg => log.warn(msg));
+    try {
+    const signal = new Int32Array(new SharedArrayBuffer(4));
+      this.worker.postMessage({ signal, port: ioChannel.port1, workerOptions }, [
+        ioChannel.port1
+      ]);
 
-    return css;
+      Atomics.wait(signal, 0, 0);
+
+      const { css, warnings } = receiveMessageOnPort(ioChannel.port2).message;
+
+      warnings.forEach(msg => log.warn(msg));
+      return css;
+    } catch (error) {
+      throw error;
```

Comment:
Such catch, much wow?

Reply:
😁

Reply:
Forgot that we can use finally without catch